docs/developer/implementing-a-feature.md

```diff
@@ -0,0 +1,61 @@
+# Implementing a Feature
+
+This document provides guidance on implementing new features in the project. Follow the recommended steps and best
+practices to ensure a successful feature development process.
+
+> **Note**
+>
+> If you’d like to implement a new feature, please open a discussion about the feature before creating an issue or opening a PR.
+
+1. **Assign yourself to the GitHub issue for the feature**: Each feature must have a corresponding GitHub issue to track
+   its progress.
+2. **Post any questions or comments about the feature on the corresponding issue**: This allows for better tracking and
+   visibility. If any discussions regarding the issue occur outside the issue thread, provide a summary of the
+   conversation as a comment on the issue itself. This ensures that all relevant information and discussions are
+   consolidated in one place for easy reference.
+3. **Fork the repo**: NKG follows a fork workflow, which you can learn more about in
+   the [branching and workflow](branching-and-workflow.md) documentation.
+4. **Branch**: Create a branch following the [naming conventions](branching-and-workflow.md#branch-naming-conventions).
+5. **Make changes**: Make the necessary changes for the feature.
+6. **Consider opening a draft PR**: If your feature involves substantial architecture changes, or you would like to
+   receive early feedback, consider opening a draft PR and requesting reviews from the maintainers. Draft PRs are an
+   effective means to solicit feedback and ensure that major architectural changes align with the project's goals and
+   standards.
+7. **Add or update unit tests** All features **must** be accompanied by unit tests that verify the functionality. Make
+   sure to thoroughly test the different scenarios and edge cases related to the feature to ensure robustness and
+   reliability. Additionally, open the code coverage report to ensure that the code you added has sufficient test
+   coverage. For instructions on writing and running unit tests, refer to the [testing](testing.md#unit-test-guidelines)
+   documentation.
+8. **Manually verify your changes**: Refer to the [manual testing](testing.md#manual-testing) section of the testing
+   documentation for instructions on how to manually test your changes.
+9. **Update any relevant documentation**: Here are some guidelines for updating documentation:
+    - **Gateway API Feature**: If you are implementing a Gateway API feature, make sure to update
+      the [Gateway API Compatibility](../gateway-api-compatibility.md) documentation.
```

Comment:
Is there any way to use absolute paths instead of relative? I just worry about this being a bit harder to maintain if docs move around.